src/main/java/org/sunbird/migrate/service/impl/UserMigrationServiceImpl.java

```diff
@@ -0,0 +1,247 @@
+                      } else {
+                          log.info("Organization '{}' found for user ID '{}'. No migration needed.", targetOrgName, userId);
+                      }
+                  }
+                  orgFound = false;
+              }
+
+              if (allOperationsSuccessful) {
+                  response.setResponseCode(HttpStatus.OK);
+                  response.getParams().setStatus(Constants.SUCCESS);
+              } else {
+                  response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+                  response.getParams().setStatus(Constants.FAILED);
+              }
+
+          } else {
+              // Handle error response
+              if (updateResponse != null && Constants.CLIENT_ERROR.equalsIgnoreCase((String) updateResponse.get(Constants.RESPONSE_CODE))) {
+                  Map<String, Object> responseParams = (Map<String, Object>) updateResponse.get(Constants.PARAMS);
+                  if (MapUtils.isNotEmpty(responseParams)) {
+                      String errorMessage = (String) responseParams.get(Constants.ERROR_MESSAGE);
+                      response.getParams().setErrmsg(errorMessage);
+                  }
+                  response.setResponseCode(HttpStatus.BAD_REQUEST);
+              } else {
+                  response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+              }
+              response.getParams().setStatus(Constants.FAILED);
+              String errMsg = response.getParams().getErrmsg();
+              if (StringUtils.isEmpty(errMsg)) {
+                  errMsg = (String) ((Map<String, Object>) updateResponse.get(Constants.PARAMS)).get(Constants.ERROR_MESSAGE);
+                  errMsg = PropertiesCache.getInstance().readCustomError(errMsg);
+                  response.getParams().setErrmsg(errMsg);
+              }
+              log.error(errMsg, new Exception(errMsg));
+              return response;
+          }
+      } catch (Exception e) {
+          log.error("Error during user migration: {}", e.getMessage(), e);
+          response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+          response.getParams().setStatus(Constants.FAILED);
+          response.getParams().setErrmsg(e.getMessage());
+      }
+
+        return response;
+    }
+
+
+    private Map<String, Object> userSearchRequestBody() {
+        ZoneId zoneId = ZoneId.of("UTC");
+
+        // Get the current time in UTC
+        ZonedDateTime currentTime = ZonedDateTime.now(zoneId);
+
+        // Format the current time to the desired string format
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss:SSS'+'0000");
+        String formattedCurrentTime = currentTime.format(formatter);
+
+        // Get yesterday's date at 1 AM in UTC
+        ZonedDateTime yesterdayAtOneAM = currentTime
+                .minusDays(1) // Subtract one day to get yesterday
+                .withHour(1)  // Set the hour to 1 AM
```

Comment:
don't use time
we will run this every day 1 am or between 1 am to 3 am or some other time...
so do make sure you are using currentDate - 1 -- for search... if any record is updated at 12:30am -- those would be executed next day -- since we search based on date...

Reply:
I meant not specify any hours... just keep 0 -- i.e. midnight